inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java

```diff
@@ -804,6 +804,10 @@ public Object visitExecutable(ExecutableElement method, Object o) {
                             (isExecutableThroughType(method.getEnclosingElement(), methodAnnotationMetadata, annotationMetadata, modifiers, isPublic) ||
                                     hasAroundStereotype(annotationMetadata));
 
+            if (javaMethodElement.isPrivate() && hasAroundStereotype(annotationMetadata)) {
```

Comment:
Perhaps using `isExecutable` here is more complete?

Reply:
Oh I see that variable includes `!hasInvalidModifiers`. I would say to remove that condition then check it inside the `if (isDeclaredBean && isExecutable) {` condition. This check also needs done for Groovy

Reply:
I had initially replicated this in Groovy AST code, but private methods don't show up as private.